server/master/src/main/java/org/apache/accumulo/master/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -328,38 +333,45 @@ private long loadFiles(Table.ID tableId, Path bulkDir, LoadMappingIterator lmi,
     loader.start(bulkDir, master, tid, bulkInfo.setTime);
 
     long t1 = System.currentTimeMillis();
-    while (true) {
-      if (loadMapEntry == null) {
-        if (!lmi.hasNext()) {
-          break;
-        }
-        loadMapEntry = lmi.next();
-      }
-      KeyExtent fke = loadMapEntry.getKey();
-      Files files = loadMapEntry.getValue();
-      loadMapEntry = null;
-
+    while (loadMapEntry != null) {
+      KeyExtent fileTablet = loadMapEntry.getKey();
       tablets.clear();
 
-      while (!Objects.equals(currentTablet.getPrevEndRow(), fke.getPrevEndRow())) {
-        currentTablet = tabletIter.next();
-      }
+      // get first tablet in range
+      currentTablet = getFirstTablet(tabletIter, currentTablet, fileTablet.getPrevEndRow());
       tablets.add(currentTablet);
 
-      while (!Objects.equals(currentTablet.getEndRow(), fke.getEndRow())) {
-        currentTablet = tabletIter.next();
-        tablets.add(currentTablet);
-      }
+      // get all tablets between currentTablet and EndRow given in mapping file
```

Comment:
Personally I think these loops inlined are easier to understand.  The method names do not give me any concrete idea what the method will do with the parameters.  So I still need to go look at the method impl to understand the big picture.  

I think the loops inlined with comments you put before the methods would be good.

Reply:
Those names would be better.  The words `skip` and `advance` are other possible method name prefixes.